RecoTracker/TkSeedGenerator/plugins/ElectronSeedFitter.cc

```diff
@@ -0,0 +1,281 @@
+
+  // initialize the "event dependent state"
+  void init(const edm::EventSetup& es);
+
+  // make job
+  // fill seedCollection with the "ElectronSeedCollection"
+  void makeSeed(reco::ElectronSeedCollection& seedCollection, const reco::ElectronSeed& seed);
+
+private:
+  ElectronSeedFitter(const edm::ParameterSet&, edm::ConsumesCollector&&);
+
+  void initialKinematic(GlobalTrajectoryParameters& kine, const reco::ElectronSeed& seed) const;
+
+  CurvilinearTrajectoryError initialError(float sin2Theta) const dso_hidden;
+
+  void buildSeed(reco::ElectronSeedCollection& seedCollection,
+                 const reco::ElectronSeed& seed,
+                 const FreeTrajectoryState& fts) const dso_hidden;
+
+  SeedingHitSet::RecHitPointer refitHit(SeedingHitSet::ConstRecHitPointer hit,
+                                        const TrajectoryStateOnSurface& state) const dso_hidden;
+
+protected:
+  const edm::EDGetTokenT<reco::ElectronSeedCollection> eleSeedCollectionToken_;
+
+  const std::string propagatorLabel_;
+  const float bOFFMomentum_;
+  const float originTransverseErrorMultiplier_;
+  const float minOneOverPtError_;
+
+  TrackerGeometry const* trackerGeometry_;
+  Propagator const* propagator_;
+  MagneticField const* magneticField_;
+  float nomField_ = 0.;
+  bool isBOFF_ = false;
+  const std::string ttrhBuilder_;
+  const std::string mfName_;
+
+  TkClonerImpl cloner_;
+
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> trackerGeometryESToken_;
+  const edm::ESGetToken<Propagator, TrackingComponentsRecord> propagatorESToken_;
+  const edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> magneticFieldESToken_;
+  const edm::ESGetToken<TransientTrackingRecHitBuilder, TransientRecHitRecord> transientTrackingRecHitBuilderESToken_;
+};
+
+ElectronSeedFitter::ElectronSeedFitter(const edm::ParameterSet& cfg)
+    : ElectronSeedFitter::ElectronSeedFitter(cfg, consumesCollector()) {}
+
+ElectronSeedFitter::ElectronSeedFitter(const edm::ParameterSet& cfg, edm::ConsumesCollector&& iC)
+    : eleSeedCollectionToken_(
+          consumes<reco::ElectronSeedCollection>(cfg.getParameter<edm::InputTag>("eleSeedCollection"))),
+      propagatorLabel_(cfg.getParameter<std::string>("propagator")),
+      bOFFMomentum_(cfg.getParameter<double>("SeedMomentumForBOFF")),
+      originTransverseErrorMultiplier_(cfg.getParameter<double>("OriginTransverseErrorMultiplier")),
+      minOneOverPtError_(cfg.getParameter<double>("MinOneOverPtError")),
+      ttrhBuilder_(cfg.getParameter<std::string>("TTRHBuilder")),
+      mfName_(cfg.getParameter<std::string>("magneticField")),
+      trackerGeometryESToken_(iC.esConsumes()),
+      propagatorESToken_(iC.esConsumes(edm::ESInputTag("", propagatorLabel_))),
+      magneticFieldESToken_(iC.esConsumes(edm::ESInputTag("", mfName_))),
+      transientTrackingRecHitBuilderESToken_(iC.esConsumes(edm::ESInputTag("", ttrhBuilder_))) {
+  produces<reco::ElectronSeedCollection>();
```

Comment:
Can you store this in an `edm::EDPutTokenT<reco::ElectronSeedCollection>` and use the token in the call to `iEvent.put()` or `iEvent.emplace()` ?